kin-sdk-core/src/main/java/kin/sdk/core/KinAccountImpl.java

```diff
@@ -60,20 +71,34 @@ public String exportKeyStore(String passphrase, String newPassphrase) throws Pas
     @Override
     public TransactionId sendTransactionSync(String publicAddress, String passphrase, BigDecimal amount)
         throws InsufficientBalanceException, OperationFailedException, PassphraseException {
+        if (hasDeleted) {
+            throw new AccountDeletedOpreationFailedException();
+        }
         return ethClient.sendTransaction(account, passphrase, publicAddress, amount);
     }
 
     @Override
     public Balance getBalanceSync() throws OperationFailedException {
+        if (hasDeleted) {
+            throw new AccountDeletedOpreationFailedException();
+        }
         return ethClient.getBalance(account);
     }
 
     @Override
     public Balance getPendingBalanceSync() throws OperationFailedException {
+        if (hasDeleted) {
+            throw new AccountDeletedOpreationFailedException();
+        }
         return ethClient.getPendingBalance(account);
     }
 
     void delete(String passphrase) throws DeleteAccountException {
         ethClient.deleteAccount(account, passphrase);
+        markAsDeleted();
+    }
+
+    public void markAsDeleted() {
```

Comment:
should be package private (for consistency, though it's meaningless as the whole class is package private)